### PR TITLE
feat: Implement collaborative highlight for AI-modified elements

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Implement collaborative highlight for AI-modified elements

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/canvas/collaborative-instance-react.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { useEffect } from "react";
+import { updateCollaborativeInstanceRect } from "./collaborative-instance";
+
+export const useUpdateCollaborativeInstanceRect = () => {
+  useEffect(updateCollaborativeInstanceRect, []);
```
Comment (TrySound) [inline apps/builder/app/canvas/collaborative-instance-react.ts:5]: Just put it into canvas, too much repetition.

Comment (istarkov) [reply apps/builder/app/canvas/collaborative-instance-react.ts:5]: ?

File: apps/builder/app/builder/features/ai/ai-command-bar.tsx [hunk 2]
```diff
@@ -172,7 +177,21 @@ export const AiCommandBar = ({ isPreviewMode }: { isPreviewMode: boolean }) => {
...
     // Skip Abort Logic for now
     try {
-      await fetchResult(prompt, abortController.current.signal);
+      const instanceSelector = (
+        selectedInstanceSelectorStore.get() ?? [
+          selectedPageStore.get()?.rootInstanceId,
+        ]
+      ).filter(<T,>(item: T): item is NonNullable<T> => item !== undefined);
+
+      const instanceId = instanceSelector[0];
```
Comment (TrySound) [inline apps/builder/app/builder/features/ai/ai-command-bar.tsx:186]: const [instanceId] = instanceSelector

File: apps/builder/app/builder/features/ai/ai-command-bar.tsx [hunk 3]
```diff
@@ -172,7 +177,21 @@ export const AiCommandBar = ({ isPreviewMode }: { isPreviewMode: boolean }) => {
 
     // Skip Abort Logic for now
     try {
-      await fetchResult(prompt, abortController.current.signal);
+      const instanceSelector = (
+        selectedInstanceSelectorStore.get() ?? [
+          selectedPageStore.get()?.rootInstanceId,
+        ]
+      ).filter(<T,>(item: T): item is NonNullable<T> => item !== undefined);
```
Comment (TrySound) [inline apps/builder/app/builder/features/ai/ai-command-bar.tsx:184]: This a little more readable.
```ts
const page = selectedPageStore.get()
const rootInstanceSelector = page ? [page.rootSelectorId] : []
const instanceSelector = selectedInstanceSelectorStore.get() ?? rootInstanceSelector
```

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/canvas/collaborative-instance-react.ts (hunk 1 above)
Comment (TrySound) [reply apps/builder/app/canvas/collaborative-instance-react.ts:5]: I mean see where you use the hook. Just add `useEffect(hook, [])` on canvas

**Pull request merged**